### PR TITLE
Client side needs a plumb endpoint component

### PR DESCRIPTION
**Pull request by justinvdm**
jsPlumb provides `Endpoint` objects as for one side of a connection (or a placeholder for one). We need a wrapper component for these objects in order to make them work nicer with Backbone Models and Views, and to provide some re-useable functionality for the different parts of the client side that need plumbing.

### CODE REVIEW

File: go/base/static/js/src/utils.js [hunk 1]
```diff
@@ -16,4 +16,33 @@
...
+      for (var e in events) { that.on(e, that[events[e]].bind(that)); }
+  };
+
+  // Returns the internal prototype ([[Prototype]]) of the instance's internal
+  // prototype (One step up in the prototype chain, and thus the 'super'
+  // prototype).
+  //
+  // If `propName` is specified, a property on 'super' prototype is returned.
+  // If the property is a function, the function is bound to the instance.
+  exports._super_ = function(that, propName) {
```
Comment (justinvdm) [inline go/base/static/js/src/utils.js]: Any idea for a better name than `_super_` (`super` is reserved)?

Comment (peterp) [inline go/base/static/js/src/utils.js]: What about `parent`?

Comment (justinvdm) [inline go/base/static/js/src/utils.js]: `parent` it is, thanks! :)

**Comment by justinvdm** [comment]
I'm making quite a few changes to this PR, not worth keeping it open review, but will have it re-opened soon.

**Comment by peterp** [comment]
Other than my comment above :+1:

**Comment by justinvdm** [comment]
I'm refactoring some things and moving some out of the plumbing component, so I'd rather close this PR again for now (sorry for constantly opening and closing this PR). On the bright side: as a side effect of these changes, it will make this PR a bit smaller and more manageable for reviewing.

**Pull request closed without merging**